src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Debugging/RazorProximityExpressionsEndpoint.cs

```diff
@@ -8,39 +8,28 @@
 using Microsoft.AspNetCore.Razor.Language;
 using Microsoft.AspNetCore.Razor.LanguageServer.Common.Extensions;
 using Microsoft.AspNetCore.Razor.LanguageServer.EndpointContracts.Debugging;
-using Microsoft.AspNetCore.Razor.LanguageServer.Extensions;
-using Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
 using Microsoft.AspNetCore.Razor.LanguageServer.Protocol;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.ExternalAccess.Razor;
-using Microsoft.CodeAnalysis.Razor;
-using Microsoft.CodeAnalysis.Razor.ProjectSystem;
 using Microsoft.CodeAnalysis.Text;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Debugging
 {
     internal class RazorProximityExpressionsEndpoint : IRazorProximityExpressionsEndpoint
     {
-        private readonly ProjectSnapshotManagerDispatcher _projectSnapshotManagerDispatcher;
```

Comment:
Most changes to an Endpoint should be almost exactly like this. Remove the `ProjectSnapshotManagerDispatcher` and `DocumentResolver`, replace it with `DocumentContextFactory`, change the appropriate method calls and result usage and you're done. There were a couple places where `ProjectSnapshotManagerDispatcher` got left around because there's changes being made to the document state. If that's still undesirable let me know and I can work to eliminate all uses of it.